### PR TITLE
Refactor osgi test

### PR DESCRIPTION
**Pull request by dmitryintel**
* Move all the unit tests to NeutronSfcSdnRedirectionApiTest.
* Use InMemDB
* Two minimal tests retained in OSGiIntegrationTest, to show OSGi setup is working.

### CODE REVIEW

File: src/main/resources/META-INF/persistence.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <persistence xmlns="http://java.sun.com/xml/ns/persistence"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-   xsi:schemaLocation="http://java.sun.com/xml/ns/persistence http://java.sun.com/xml/ns/persistence/persistence_2_1.xsd"
-   version="2.1">
+   xsi:schemaLocation="http://java.sun.com/xml/ns/persistence http://java.sun.com/xml/ns/persistence/persistence_2_0.xsd"
```
Comment (arvindn05) [inline src/main/resources/META-INF/persistence.xml]: looks like we are using 2.1 xsd and we just need to update the links

see http://www.oracle.com/webfolder/technetwork/jsc/xml/ns/persistence/index.html#2.1

File: src/test/java/org/osc/controller/nsc/OSGiIntegrationTest.java [hunk 2]
```diff
@@ -72,24 +64,6 @@
     private static final String TEST_DB_URL_SUFFIX = ";MVCC\\=FALSE;LOCK_TIMEOUT\\=10000;MV_STORE=FALSE;";
```
Comment (arvindn05) [inline src/test/java/org/osc/controller/nsc/OSGiIntegrationTest.java]: do we need to use DB file? can we swicth to using the inmem db here as well?

File: src/test/java/org/osc/controller/nsc/SampleSdnRedirectionApiTestData.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+import static java.util.Arrays.asList;
+
+import org.osc.controller.nsc.entities.InspectionHookEntity;
+import org.osc.controller.nsc.entities.InspectionPortEntity;
+import org.osc.controller.nsc.entities.NetworkElementEntity;
+
+public class SampleSdnRedirectionApiTestData {
+
+    public static final String EADDR2_STR = "192.168.0.12";
```
Comment (arvindn05) [inline src/test/java/org/osc/controller/nsc/SampleSdnRedirectionApiTestData.java]: make these private or package private

File: src/test/java/org/osc/controller/nsc/SampleSdnRedirectionApiTest.java [hunk 4]
```diff
@@ -0,0 +1,638 @@
...
+        setupDataObjects();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        InMemDB.close();
+    }
+
+    @Test
+    public void verifyCorrectNumberOfMacsAdPortIps() throws Exception {
```
Comment (arvindn05) [inline src/test/java/org/osc/controller/nsc/SampleSdnRedirectionApiTest.java]: assume all of this code is moved from the previous integration test?

Any hot spots to look for?

Also is the plan to update them to the arrange, act and assert as part of next checkin?

**Comment by arvindn05** [comment]
Is Jacoco going to be integrated as part of this change? We can enable code cov for this repo as well once jacoco is integrated as part of pom

File: pom.xml [hunk 5]
```diff
@@ -531,6 +546,27 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.jacoco</groupId>
+                <artifactId>jacoco-maven-plugin</artifactId>
+                <version>0.7.9</version>
+
```
Comment (arvindn05) [inline pom.xml]: remove extra line

File: src/test/java/org/osc/controller/nsc/OSGiIntegrationTest.java [hunk 6]
```diff
@@ -67,28 +58,7 @@
...
-    private static final String EMAC2_STR = "ee:ff:aa:bb:cc:02";
-
-    private static final String EMAC1_STR = "ee:ff:aa:bb:cc:01";
-
-    private static final String IMAC1_STR = "ff:ff:aa:bb:cc:01";
-
-    private static final String IMAC2_STR = "ff:ff:aa:bb:cc:02";
-
-    private static final String INSPMAC1_STR = "aa:aa:aa:bb:cc:01";
+    private static final String TEST_DB_URL = "jdbc:h2:mem:nsfc-mgr;DB_CLOSE_DELAY=-1";
```
Comment (arvindn05) [inline src/test/java/org/osc/controller/nsc/OSGiIntegrationTest.java]: why do we want to keep the content of the db until the JVM shutsdown?

also DB name should be "nsc-osgi-db" not "nsfc"

also not used private field? Should result in eclipse warning...please make sure no warnings are introduced

File: src/test/java/org/osc/controller/nsc/SampleSdnRedirectionApiTest.java [hunk 7]
```diff
@@ -0,0 +1,638 @@
...
+        });
+
+        assertEquals(2, tmp.getEgressPort().getMacAddresses().size());
+        assertEquals(2, tmp.getEgressPort().getPortIPs().size());
+        assertEquals(2, tmp.getIngressPort().getMacAddresses().size());
+        assertEquals(2, tmp.getIngressPort().getPortIPs().size());
+    }
+
+    @Test
+    public void verifyHookAndPortPersistedAfterSingleHookPersistenceWithObjectGraphSetUp() {
```
Comment (arvindn05) [inline src/test/java/org/osc/controller/nsc/SampleSdnRedirectionApiTest.java]: doesnt seem like this is a test of sdn redirection api...more like the em tests. We should split to a different test suite/file

File: src/test/java/org/osc/controller/nsc/SampleSdnRedirectionApiTest.java [hunk 8]
```diff
@@ -0,0 +1,638 @@
...
+        assertEquals(inspectionHook.getHookId(), persistedHook.getHookId());
+
+        InspectionPortEntity persistedPort = persistedHook.getInspectionPort();
+
+        assertNotNull(persistedPort);
+        assertEquals(inspectionHook.getInspectionPort().getElementId(), persistedPort.getElementId());
+    }
+
+    @Test
+    public void testUtilsInspPortByNetworkElements() throws Exception {
```
Comment (arvindn05) [inline src/test/java/org/osc/controller/nsc/SampleSdnRedirectionApiTest.java]: this tests utils...should move to RedirectionUtilsTest class

File: src/test/java/org/osc/controller/nsc/SampleSdnRedirectionApiTest.java (hunk 4 above)
Comment (dmitryintel) [reply src/test/java/org/osc/controller/nsc/SampleSdnRedirectionApiTest.java]: This was a simple copy, yes no actual tests were changed, except tearDown code and DB url.

File: src/test/java/org/osc/controller/nsc/OSGiIntegrationTest.java (hunk 6 above)
Comment (dmitryintel) [reply src/test/java/org/osc/controller/nsc/OSGiIntegrationTest.java]: I want to be the one who clears the database. Without the DB_CLOSE_DELAY, the tests would not pass. em.close() does not work.

Renamed the databases.

redirApi? It was overshadowed. Fixed that.

**Pull request merged**